internal/core/src/query/PlanImpl.h

```diff
@@ -64,19 +64,30 @@ struct Plan {
 struct Placeholder {
     std::string tag_;
     int64_t num_of_queries_;
-    int64_t line_sizeof_;
+    // TODO(SPARSE): add a dim_ field here, use the dim passed in search request
```

Comment:
is there a special reason we need this dim in search?

Reply:
The main reason is to make sure the input for knowhere conforms to the API invariant: the `dataset->GetDim()` correctly returns the dim of the contained data, which is not yet true for those sparse query dataset. Though knowhere doesn't use the dim of the query dataset now.